go/logic/applier.go

```diff
@@ -986,13 +990,23 @@ func (this *Applier) ApplyDMLEventQueries(dmlEvents [](*binlog.BinlogDMLEvent))
 	var totalDelta int64
 
 	err := func() error {
+		if !this.migrationContext.UseBinLog {
+			if _, err := sqlutils.ExecNoPrepare(this.db, `SET sql_log_bin = OFF`); err != nil {
```

Comment:
This statement happens within a transaction so I need to selectively turn it off/on again outside the transaction.

Reply:
This won't do what you expect... `this.db` is a `sql.DB` here right? If so, it represents a connection pool, not a single connection. There's no guarantee that the connection used for `SET sql_log_bin = OFF` will be the same connection used for `tx, err := this.db.Begin()`... you'd need to use [this.db.Conn()](https://golang.org/pkg/database/sql/#DB.Conn) here too I believe.

Reply:
Thanks @evanelias!